### ISSUE TITLE
No dangerous-default-warning with keyword-only arguments 

### ISSUE THREAD

**Issue by andreaswachowski** [PyCQA/pylint#3373]
### Steps to reproduce
1. Create a file with the following contents:
```
class SomeClass:
    def __init__(self, *, a_list=[]):
        self.a_list = a_list
```
2. Run pylint on this file.

### Current behavior

pylint does not raise the dangerous-default-value-warning. (I might be missing something, but shouldn't it?)

### Expected behavior

pylint raises (among others) the dangerous-default-value-warning:

```
W0102: Dangerous default value [] as argument (dangerous-default-value)
```

Note: That warning _is_ raised when the keyword-only indicator `*` is removed.

### pylint --version output

pylint 2.4.4
astroid 2.3.3
Python 3.7.6 (default, Jan  4 2020, 10:59:20)
[Clang 11.0.0 (clang-1100.0.33.12)]

**Comment by PCManticore** [comment]
Thanks, we should be able to emit a warning for that as well.

**Comment by andreaswachowski** [comment]
Thanks! Short follow-up, the bug can be reproduced with this simplified example:
```
def f(*, a=[]):
    pass
```

whereas

```
def f(a=[]):
    pass
```

works fine.
